content/3_Silver/Binary_Search.mdx

```diff
@@ -1169,3 +1169,181 @@ for _ in range(queryNum):
 ### General
 
 <Problems problems="general" hideSuggestProblemButton />
+
+## Quiz
+				
+<Quiz>
+  <Quiz.Question>
+    What is the worst and the best case time complexity for binary searching for a number in a sorted array of size $n$?
+    <Quiz.Answer>
+      $\mathcal{O}(\log n)$ / $\Omega(\log n)$
+      <Quiz.Explanation>
+        Incorrect. Consider the case where the target element is the median of the array. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      $\mathcal{O}(\log n)$ / $\Omega(1)$
+      <Quiz.Explanation>
+        Correct. The worst case happens when the number is not present, and the best case occurs if the first number we check is the target value. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $\mathcal{O}(n)$ / $\Omega(\log n)$
+      <Quiz.Explanation>
+        Incorrect. Binary search always takes at most $\lceil \log n \rceil$ iterations. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      $\mathcal{O}(\log n)$ / $\Omega(\log \log n)$
+      <Quiz.Explanation>
+        Incorrect. Consider the case where the target element is the median of the array. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    What is an advantage of using binary search over linear search?
+    <Quiz.Answer correct>
+      Faster time complexity
+      <Quiz.Explanation>
+        Correct. Linear search runs in $\mathcal{O}(n)$, whereas binary search runs in $\mathcal{O}(\log n)$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Less Memory
+      <Quiz.Explanation>
+        Incorrect. Binary search and linear search both use $\mathcal{O}(n)$ memory.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Easier implementation. 
+      <Quiz.Explanation>
+        Incorrect. Linear search is, in most cases, a simple `for` loop, whereas binary search most often requires at least two additional variables and conditionals. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Suppose a program involving binary search runs in $\mathcal{O}(\log n)$ time for some $n$. At what values of $n$ would runtime be an issue? A typical online judge can handle around $10^9$ operations in a second. 
```

Comment:
probably best to reword to "what is the lowest value of N" where runtime would be an issue otherwise just guess highest number (and then maybe change the answers accordingly)